resources/telemetry.md

```diff
@@ -0,0 +1,369 @@
+---
+layout: default
+nav_order: 3
+parent: Resources
+grand_parent: Maintainer Docs
+title: Telemetry
+---
+
+# {{ page.title }}
+{:.no_toc}
+
+## Overview
+{:.no_toc}
+
+The RAPIDS team collects build-time and test-time telemetry from our CI jobs.
+These data are used to provide insights into potential speed impacts and
+optimizations that might be worthwhile.
+
+### Intended audience
+{: .no_toc }
+
+Operations
+{: .label .label-purple}
+
+Developers
+{: .label .label-green}
+
+## Table of contents
+{: .no_toc .text-delta }
+
+1. TOC
+{:toc}
+
+## Introduction
+
+This is an overview of how telemetry data is collected, how it is transmitted and recorded, and how visualizations can be obtained using Grafana.
+
+## Collecting data
+
+We can collect telemetry data from possibly multiple sources. Right now, the
+main source of timing information is the Github Actions job metadata, which we
+obtain using the [Github Actions REST
+API](https://docs.github.com/en/rest/actions/workflow-jobs?apiVersion=2022-11-28#list-jobs-for-a-workflow-run-attempt).
+These data contain jobs, which generally correspond to calls to reusable
+workflows from our shared-workflows repository. They also itemize each step
+within any given workflow.
+
+The general implementation for telemetry is as follows:
+
+1. Creating initial variables that get loaded
+2. add metadata in shared-workflows using OTEL_ATTRIBUTES, run workflow, stash variables to collect in final job
+3. On a self-hosted node, parse Github Actions job metadata, and associate jobs with attributes passed in from step 2). Send data to Tempo via a FluentBit agent that is also used to aggregate logs.
+
+Steps 1 and 3 happen as jobs in the "top-level repository" - the workflow that initiates all subsequent workflows.
+
+### Code changes for projects to add telemetry
+
+These are the general steps necessary to add telemetry to a repository that does
+not currently have it. This does not describe changes to shared-workflows as
+implementation details. In the top-level workflow, such as
+`cudf/.github/workflows/pr.yaml`:
+
+1. Add a job for telemetry-setup, and add that job name to the pr-builder `needs` collection.
+```
```

Comment:
Need a newline before the triple backticks because the code-block isn't getting rendered correctly.